### DISCUSSION TITLE
How to use py2exe.freeze

### DISCUSSION THREAD

**Discussion by tugiyo523** [py2exe/py2exe#180] [General]
Hi,

I tried to compile my .py following this: http://www.py2exe.org/index.cgi/Tutorial#Step4

but fail: 

base) D:\Cordex\machine learning\tes2>python setup.py py2exe
...... long lines .....
 File "<frozen importlib._bootstrap_external>", line 1342, in find_spec
  File "<frozen importlib._bootstrap_external>", line 1314, in _get_spec
  File "<frozen importlib._bootstrap_external>", line 1469, in find_spec
  File "<frozen importlib._bootstrap_external>", line 101, in _path_isfile
  File "<frozen importlib._bootstrap_external>", line 93, in _path_is_mode_type
  File "<frozen importlib._bootstrap_external>", line 87, in _path_stat
RecursionError: maximum recursion depth exceeded while calling a Python object

Any suggestions?
Thanks

**Comment by tugiyo523** [comment]
Installation
pip install py2exe

Usage
Use the py2exe.freeze function as documented [here](https://github.com/py2exe/py2exe/blob/master/docs/py2exe.freeze.md).

freeze(
    console=[],
    windows=[],
    data_files=None,
    zipfile='library.zip',
    options={},
    version_info={}
)

What 'py2exe.freeze' mean? how to implement it ?

I want to compile my .py with name scihubku.py

Any suggestions?
Thanks

**Comment by tugiyo523** [comment]
I tried:

from distutils.core import setup
import py2exe

freeze(
    #console=['scihubku.py'],
    windows= ['scihubku.py']
    )

but fail:
(base) D:\Cordex\machine learning\tes2>python setup2.py py2exe
Traceback (most recent call last):
  File "setup2.py", line 5, in <module>
    freeze(
NameError: name 'freeze' is not defined

**Comment by LemonzPCMR** [comment]
Cant seem to find the right format for options or something else is broken

freeze(
#console=[],
windows=['main.py'],
data_files=None,
zipfile='library.zip',
options={"bundle_files" == 1,},
version_info={}
)

line 143, in freeze
    compress = options.get("compressed", 0),
AttributeError: 'set' object has no attribute 'get'

**Comment by LemonzPCMR** [comment]
For those struggling with format like I was here you go

freeze(
#console=[],
windows=['main.py'],
data_files=None,
zipfile='library.zip',
options = {'py2exe': {'bundle_files': 1, 'compressed': True}},
version_info={'version': '0.0.0', 'product_name': 'example'}
